### ISSUE TITLE
Cuda

### ISSUE THREAD

**Issue by hadilq** [p00f/clangd_extensions.nvim#63]
Apparently [clangd is supporting CUDA](https://clangd.llvm.org/faq#does-clangd-support-cuda) but I cannot figure out how to enable it in Neovim with `clangd_extensions.nvim`. Any suggestion?

**Comment by p00f** [comment]
How are you starting clangd?

**Comment by hadilq** [comment]
I have the following configs 
```
require("clangd_extensions").setup({})

local function on_attach(client, bufnr)
  local opts = {buffer = bufnr, remap = false}
  vim.keymap.set("n", "<leader>vrn", function() vim.lsp.buf.rename() end, opts)
 --  ....
end

local lspconfig = require('lspconfig')
local capabilities = require('cmp_nvim_lsp').default_capabilities()

lspconfig.clangd.setup({
  on_attach = on_attach,
  capabilities = capabilities,
  filetypes = { "c", "cpp", "objc", "objcpp", "cuda", "proto" },
})

lspconfig.clangd_extensions.setup({
  on_attach = on_attach,
  capabilities = capabilities,
  filetypes = { "c", "cpp", "objc", "objcpp", "cuda", "proto" },
})
```

and also 
```
CompileFlags:
Add:
   - -std=c++14
   - --cuda-path=/nix/store/0c8nf26hx9x9jxgj0s9bq10xg75nbfv0-cuda-merged-12.2
   - -xcuda
   - --cuda-gpu-arch=sm_89
   - -L/nix/store/0c8nf26hx9x9jxgj0s9bq10xg75nbfv0-cuda-merged-12.2/lib
   - -I/nix/store/0c8nf26hx9x9jxgj0s9bq10xg75nbfv0-cuda-merged-12.2/include
```
in the root of the project. but apparently it's not running! For instance when I try `:ClangdAST` it's doing nothing! even no error! How can I make it work?

**Comment by hadilq** [comment]
It started to work!
```

local function on_attach(client, bufnr)
  local opts = {buffer = bufnr, remap = false}

  vim.keymap.set("n", "<leader>vrn", function() vim.lsp.buf.rename() end, opts)
  -- ....

  require("clangd_extensions.inlay_hints").setup_autocmd()
  require("clangd_extensions.inlay_hints").set_inlay_hints()
end

require("clangd_extensions").setup({})

local lspconfig = require('lspconfig')
local capabilities = require('cmp_nvim_lsp').default_capabilities()

lspconfig.clangd.setup({
  on_attach = on_attach,
  capabilities = capabilities,
  filetypes = { "c", "cpp", "objc", "objcpp", "cuda", "proto" },
})

```

It works on CUDA the same as cpp and c file :tada: Thank you for your great work.

it's obvious that I am using nix, however, I had not added `clang-tools` to the dependencies, so there was no `clangd` available, but `clang_extension` didn't provide me with any error. So hope this feedback is helpful.
